### PR TITLE
Routing branch

### PR DESCRIPTION
**Pull request by ChoudharyHamza**
added the routing feature to the project using Vue router

### CODE REVIEW

File: src/components/RenderBlog.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    axios.get("https:/jsonplaceholder.typicode.com/posts")
+      .then(response => {this.respons(response);this.paginationAlgo(this.activePage);})
+      .catch(error => { this.errorr = true;  });
+  },
+
+  data(){
+    return {
+      posts : [],
+      errorr:false,
+      returnedEmpty:false,
```
Comment (ayousaf11) [inline src/components/RenderBlog.vue]: @ChoudharyHamza this `data` is not being used. You can remove it.

File: src/components/RenderBlog.vue [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      arrToRender:[],
+
+
```
Comment (ayousaf11) [inline src/components/RenderBlog.vue]: @ChoudharyHamza remove these extra lines

File: src/components/postPagination.vue [hunk 3]
```diff
@@ -1,14 +1,18 @@
 <template>
-    <div class="btn-con">
+    <div class= "btn-con">
         {{this.buttonsRen()}}
-    <button class="btn" v-for="button in buttons" @click="paginationAlgo(button)" :class="currentPage(button)" :key="button">{{ button }}</button>
-        </div>
+        <button class= "btn" v-for= "button in buttons"  :key= "button">
```
Comment (ayousaf11) [inline src/components/postPagination.vue]: @ChoudharyHamza make this whole button clickable

File: src/main.js [hunk 4]
```diff
@@ -1,8 +1,24 @@
...
+import RenderBlog from "./components/RenderBlog"
+import PostDetail from "./components/PostDetail"
 
 Vue.config.productionTip = false
 
+Vue.use(VueRounter);
+
+let routes = [
+  { path: "/:activePage", name: "RenderBlogWithPage", component: RenderBlog, props: true },
+  { path: "/", name: "RenderBlog" ,component: RenderBlog,},
```
Comment (ayousaf11) [inline src/main.js]: @ChoudharyHamza make these as one route

Comment (ChoudharyHamza) [reply src/main.js]: all changes done

File: src/components/PostDetail.vue [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    computed: {
+        checkIds(){
+          return this.postId==this.currentPost.id;
+        },
+    },
+    watch:{
+        checkError(){
+        alert("boom!  errorr tis an errorr");
+        }
+    }
```
Comment (ayousaf11) [inline src/components/PostDetail.vue]: @ChoudharyHamza when is this `watch` triggered?

Comment (ChoudharyHamza) [reply src/components/PostDetail.vue]: oh i meant it to tiggere when a porps changes and that prop was errorOccured it was changed when an API call will return false. but i forgot to add the value on which the change will trigger

**Comment by ChoudharyHamza** [comment]
added the routing branch to the master

**Pull request merged**